wrapper.py

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env python
+import time
+import sys
+from watchgod import watch
+from subprocess import Popen
+
+if  len(sys.argv) > 2 :
+    filename=sys.argv[1]
+    token=sys.argv[2]
+else:
+    print ('Please, specify a name file to run and bot token')
+    sys.exit()
+    
+
+
+p=Popen(['python3', filename, token])
+
+for changes in watch('./'):
+   print(changes)
+   if any([ './'+filename in i for i in changes]):
+      print("stoping!")
+      p.terminate()
+      time.sleep(10)
```

Comment:
You should use p.wait() instead of sleep, or sleep, then send signal 9 to force kill it and the p.wait()

Reply:
watchdog.py would be a better name and we can later extend it to restarting the bot if it crashed for example

Reply:
> You should use p.wait() instead of sleep, or sleep, then send signal 9 to force kill it and the p.wait()
In some rare cases i saw that even after the bot file terminated ,the bot itself for some reason still is active (kind of hung ). So if start the bot process again, i get a error about duplicate running bot .


So i added 10 seconds sleeping period It's short but still enough to make sure that the bot eventually is terminated.